modules/ROOT/pages/mule-high-availability-ha-clusters.adoc

```diff
@@ -13,101 +16,118 @@ Contact your customer service representative about pricing for this feature.
+* JMS topics. +
+All Mule instances connect to the same JMS topic, which might lead to repeated processing of messages when scaling the non-clustered Mule instance horizontally.
 
-* *JMS request-reply/request-response*: All Mule instances are listening for messages in the same response queue, this implies that a Mule instance might obtain a response that isn't correlated to the request it sent. This can result in incorrect responses or make a flow fail with timeout.
+* JMS request-reply/request-response. +
+All Mule instances are listening for messages in the same response queue, which implies that a Mule instance might obtain a response that isn't correlated to the request it sent and might result in incorrect responses or make a flow fail with timeout.
 
-* *Idempotent-redelivery-policy*: Idempotency doesn’t work if the same request is received by different Mule instances. Duplicated messages aren’t possible
+* Idempotent Redelivery Policy. +
+Idempotency doesn’t work if the same request is received by different Mule instances. Duplicated messages aren’t possible.
 
-* *Salesforce streaming API*: if multiple instances of the same application are deployed, they will fail since the API only supports a single consumer. No failover support in case the instance connected is stopped or crashes
+* Salesforce streaming API. +
+If multiple instances of the same application are deployed, they will fail because the API only supports a single consumer. There is no failover support in case the instance connected is stopped or crashes.
 
+== Prerequisites
 
+* You must have at least two Mule runtime engines in a cluster, each one running on different physical (or virtual) machines.
+* Mule high availability (HA) requires a reliable network connection between servers to maintain synchronization between the nodes in the cluster.
+* You must keep the following ports open to set up a Mule cluster: port `5701` and port `54327`.
+* You need to enable the multicast IP address: `224.2.2.3`. Multicast is used to discover new cluster members.
 
-== About Clustering
+== Clustering
 
-With out of the box configurations, it is recommended that you scale a cluster to no more than eight Mule runtimes. All the Mule runtimes in a cluster group together to form a single unit. Thus, you can deploy, monitor, or stop all the Mule runtimes in a cluster as if they were a single Mule runtime. If you need more nodes than eight, contact https://www.mulesoft.com/support-and-services/mule-esb-support-license-subscription[MuleSoft Support].
+With the initial configuration, MuleSoft recommends that you scale a cluster to no more than eight Mule runtime engines. All the Mule runtime engines in a cluster group together to form a single unit. Thus, you can deploy, monitor, or stop all the Mule runtime engines in a cluster as if they were a single Mule runtime engine.
```

Comment:
Are these features of ARM? (deploy, monitor, stop all runtimes in a cluster)

Reply:
This is also in the 3.9 article, so this was just moved along to 4.1 but can't confirm if it is referring to ARM.

Reply:
Ok, just want to confirm because these features are not natively supported by Mule, but maybe ARM provides them.

Reply:
Who is able to confirm this?

Reply:
Addressed in commit f73725d